src/MediaRental/DatabaseSupport.java

```diff
@@ -8,459 +9,568 @@
-				"PRIMARY KEY (id), " +
-				"FOREIGN KEY (transactionID) REFERENCES Transaction(id), " +
-				"FOREIGN KEY (productID) REFERENCES Product(id));";
-
-		try
-		{
-			Statement stmt = conn.createStatement();
-			stmt.execute(statement);
-			stmt.execute(statement2);
-			stmt.execute(statement3);
-			stmt.execute(statement4);
-			stmt.execute(statement5);
-			stmt.execute(statement6);
-			stmt.close();
-			return true;
-		} catch (SQLException E)
-		{
-			System.out.println("SQLException: " + E.getMessage());
-			System.out.println("SQLState: " + E.getSQLState());
-			System.out.println("VendorError: " + E.getErrorCode());
-			return false;
-		}
-
-	}
+    public static final String DB_URL = "jdbc:mysql://127.0.0.1:3306/movie_rental";
+    public static final String USER = "root";
+    public static final String PASSWORD = "pass";
+    public static final String DB_DRIVER = "com.mysql.jdbc.Driver";
+    private static Connection conn;
+
+    public DatabaseSupport()
+    {
+        try
+        {
+            Class.forName(DB_DRIVER);
+        } catch (Exception E)
+        {
+            System.err.println("Unable to load driver.");
+            E.printStackTrace();
+        }
+        try
+        {
+            conn = DriverManager.getConnection(DB_URL, USER, PASSWORD);
+        } catch (SQLException E)
+        {
+            System.out.println("SQLException: " + E.getMessage());
+            System.out.println("SQLState: " + E.getSQLState());
+            System.out.println("VendorError: " + E.getErrorCode());
+        }
+    }
+
+    /**
+     * @param title - can be null
+     * @param genre - can be null
+     * @return customers that have a name or address similar to the passed in values
+     */
+    public ArrayList<Product> findProducts(String title, String genre)
+    {
+        String statement = "Select id, title, genre, description, catalogID from ProductCatalog";
+        String whereClause = "";
+        ArrayList<Product> products = new ArrayList<Product>();
+        if (title != null && !title.isEmpty())
+        {
+            whereClause += "where title like %'" + title + "%'";
+            if (genre != null && !genre.isEmpty())
+            {
+                whereClause += "and genre like %'" + genre + "%'";
+            }
+        } else if (genre != null && !genre.isEmpty())
+        {
+            whereClause += "where genre like %'" + genre + "%'";
+        }
+        statement += whereClause + ";";
+        try
+        {
+            Statement stmt1 = conn.createStatement();
+            ResultSet rs1 = stmt1.executeQuery(statement);
+            while (rs1.next())
+            {
+                int id = rs1.getInt("id");
+                String t = rs1.getString("title");
+                String g = rs1.getString("genre");
+                String desc = rs1.getString("description");
+                Product prod = new Product(title, "", g, desc, id);
+                prod.setGenre(g);
```

Comment:
I don't think we still need to use these setters if we set them in the constructor